src/Activities/Internal/ActivityMetricsSender.cs

```diff
@@ -21,70 +18,61 @@ internal sealed class ActivityMetricsSender : IActivitiesEventSender, IDisposabl
 		private readonly Histogram<long> m_healthCheckActivityHistogram;
 		private readonly IExecutionContext m_context;
 		private readonly IHostEnvironment m_hostEnvironment;
-		private readonly ArrayPool<KeyValuePair<string, object?>> m_arrayPool;
+		private readonly HashSet<string> m_extraBaggageDimension;
+		private readonly HashSet<string> m_extraTagObjectsDimension;
 
-		public ActivityMetricsSender(IExecutionContext executionContext, IHostEnvironment hostEnvironment)
+		public ActivityMetricsSender(IExecutionContext executionContext, IHostEnvironment hostEnvironment, IExtraActivityBaggageDimensions extraActivityBaggageDimensions, IExtraActivityTagObjectsDimensions extraActivityTagObjectsDimensions)
 		{
 			m_context = executionContext;
 			m_hostEnvironment = hostEnvironment;
 			m_meter = new Meter("Microsoft.Omex.Activities", "1.0.0");
 			m_activityHistogram = m_meter.CreateHistogram<long>("Activities");
 			m_healthCheckActivityHistogram = m_meter.CreateHistogram<long>("HealthCheckActivities");
-			m_arrayPool = ArrayPool<KeyValuePair<string, object?>>.Create();
+			m_extraBaggageDimension = extraActivityBaggageDimensions.ExtraDimensions;
+			m_extraTagObjectsDimension = extraActivityTagObjectsDimensions.ExtraDimensions;
 		}
 
 		public void SendActivityMetric(Activity activity)
 		{
-			long durationMs = Convert.ToInt64(activity.Duration.TotalMilliseconds);
-
-			int tagsCount = s_customTags.Length + activity.TagObjects.Count() + activity.Baggage.Count();
-
-			KeyValuePair<string, object?>[] tags = m_arrayPool.Rent(tagsCount);
```

Comment:
You maybe asking why I use `TagList` instead of ArrayPool.
1. This document explains the better performance of TagList https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics
2. The previous implementation used `int tagsCount` which take the length of Baggage before considering they match ExtraDimenion items. This design overcomes the problem of calculating length of array to rent